macros/cross_db_utils/listagg.sql

```diff
@@ -0,0 +1,50 @@
+{% macro listagg(measure, delimiter_text, order_by_clause='', limit_clause='') -%}
+  {{ return(adapter.dispatch('listagg', 'dbt_utils') (measure, delimiter_text, order_by_clause, limit_clause)) }}
+{%- endmacro %}
+
+
+{% macro default__listagg(measure, delimiter_text, order_by_clause, limit_clause) -%}
+
+    listagg(
+        {{ measure }},
+        {{ delimiter_text }}
+        )
+        {% if order_by_clause -%}
+        within group ({{ order_by_clause }})
+        {%- endif %}
+
+{%- endmacro %}
+
+{% macro bigquery__listagg(measure, delimiter_text, order_by_clause, limit_clause) -%}
+
+    string_agg(
+        {{ measure }},
+        {{ delimiter_text }}
+        {% if order_by_clause -%}
+        {{ order_by_clause }}
+        {%- endif %}
+        {% if limit_clause -%}
+        {{ limit_clause }}
+        {%- endif %}
+        )
+
+{%- endmacro %}
+
+{% macro postgres__listagg(measure, delimiter_text, order_by_clause, limit_clause) -%}
+
+    string_agg(
+        {{ measure }},
+        {{ delimiter_text }}
+        {% if order_by_clause -%}
+        {{ order_by_clause }}
+        {%- endif %}
+        )
+
+{%- endmacro %}
+
+{# redshift should use default instead of postgres #}
+{% macro redshift__listagg(measure, delimiter_text, order_by_clause, limit_clause) -%}
```

Comment:
Following the example in [datediff](https://github.com/dbt-labs/dbt-utils/blob/main/macros/cross_db_utils/datediff.sql#L62) macro - why is this necessary?

Reply:
Good question! Some background: 
- The Redshift adapter inherits from Postgres, so will try using Postgres-specific code before reverting to the default ([read more](https://docs.getdbt.com/reference/dbt-jinja-functions/dispatch#adapter-inheritance))
- Postgres' datediff implementation [is very limited and needs some ridiculous hacks](https://github.com/dbt-labs/dbt-utils/blob/main/macros/cross_db_utils/datediff.sql#L27-L58). 
- Redshift actually has a good, sensible datediff implementation of its own instead of just behaving the same way as Postgres, so we're skipping the Postgres implementation in favour of the base implementation. 

This is the right thing to do here as well - Redshift uses `listagg`, not `string_agg`, so we should use the default implementation instead of Postgres'.